baseapp/msg_service.go

```diff
@@ -0,0 +1,8 @@
+package baseapp
+
+type MsgServiceRouter struct{}
+
+// NewMsgServiceRouter creates a new MsgServiceRouter.
+func NewMsgServiceRouter() *MsgServiceRouter {
```

Comment:
the logic i see is, inside `app#runMsgs`:
- decode `tx` normally (after adding a special decode case in TxBody.UnpackInterfaces)
- for each msg in `tx.GetMsgs`
    - if msg.Route contains 2 slashes, use this new `MsgServiceRouter` router
    - if not, then use the old logic

is that what you had in mind @aaronc ?

Reply:
Also, in this struct, should we also have a `map[string]handler` map, like in the gRPC query router?